jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/ScanningAppProvider.java

```diff
@@ -58,27 +58,40 @@ public abstract class ScanningAppProvider extends ContainerLifeCycle implements
     private int _scanInterval = 10;
     private Scanner _scanner;
     private boolean _useRealPaths;
-    private String _environmentName;
     private boolean _deferInitialScan = false;
 
-    private final Scanner.DiscreteListener _scannerListener = new Scanner.DiscreteListener()
+    private final Scanner.BulkListener _scannerBulkListener = new Scanner.BulkListener()
     {
-        @Override
-        public void pathAdded(Path path) throws Exception
-        {
-            ScanningAppProvider.this.pathAdded(path);
-        }
+        private final Set<Path> _addedPaths = new HashSet<>();
 
         @Override
-        public void pathChanged(Path path) throws Exception
+        public void pathsChanged(Set<Path> paths) throws Exception
         {
-            ScanningAppProvider.this.pathChanged(path);
+            List<Path> sortedPaths = paths.stream()
+                .sorted(PathCollators.byName(true))
+                .toList();
+
```

Comment:
I was thinking that at this stage it would be best to resolve  xml vs war paths, as you can see all the paths at once.
If there is a foo.xml and/or a foo.war in the original list, then the end result should just be foo.xml to be deployed